packages/asl-metrics/scripts/migrate-rops-to-transferred-project.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @Usage: node migrate-rops-to-transferred-project.js
+ * @description: script to migrate ROPs to the transferred project, where ROPS has duplicate year submitted at Transferred project.
+ * use transferred project ROPs.
+ * */
+const db = require('@asl/schema');
+const config = require('../config');
+
+(async () => {
+  const { Project, Rop, ProjectVersion, destroy } = db(config.asldb);
+
+  try {
+    // Fetch projects with status 'transferred'
+    const result = await Project.query().select('id', 'transferProjectId').where({ status: 'transferred' });
+    const e1ProjectTransferred = result.map(project => ({
+      E1_PROJECT_ID: project.id,
+      TRANSFERRED_PROJECT_ID: project.transferProjectId
+    }));
+
+    // Count and log the results
+    const count = e1ProjectTransferred.length;
+    console.log('Count:', count);
+
+    // Process each project
+    for (const project of e1ProjectTransferred) {
```

Comment:
If this uses an async for loop over a result set stream ([see example](https://github.com/UKHomeOffice/aspel-workspace/blob/main/packages/asl-toolbox/scripts/update-protocol-locations.js#L193-L211)), or uses `await Promise.all(e1ProjectTransferred.flatMap(async (project) => {....` the updates can run in parallel and the script will complete quicker.